trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/CExpressionTranslator.java

```diff
@@ -635,7 +643,14 @@ public Result handlePrefixIncrementAndDecrement(final int prefixOp, final ILocat
 		builder.setLrValue(new RValue(valueXcremented, oType, false, false));
 		final ExpressionResult assign = funMakeAssignment.apply(builder.build());
 		final RValue tmpRValue = new RValue(auxvar.getExp(), oType);
-		return new ExpressionResultBuilder().addAllExceptLrValue(assign).setLrValue(tmpRValue).build();
+		final ExpressionResultBuilder resultBuilder = new ExpressionResultBuilder().setLrValue(tmpRValue);
+		if (oType.isAtomic()) {
+			final AtomicStatement atomic = new AtomicStatement(loc, assign.getStatements().toArray(Statement[]::new));
```

Comment:
Similar to the method above: the atomic block might contain too many statements.